jme3-core/src/main/java/com/jme3/util/BufferUtils.java

```diff
@@ -1289,22 +1289,22 @@ public static void destroyDirectBuffer(Buffer toBeDestroyed) {
      */
     private static boolean isDirect(Buffer buf) {
         if (buf instanceof FloatBuffer) {
-            return ((FloatBuffer) buf).isDirect();
+            return buf.isDirect();
         }
         if (buf instanceof IntBuffer) {
-            return ((IntBuffer) buf).isDirect();
+            return buf.isDirect();
         }
         if (buf instanceof ShortBuffer) {
-            return ((ShortBuffer) buf).isDirect();
+            return buf.isDirect();
         }
         if (buf instanceof ByteBuffer) {
-            return ((ByteBuffer) buf).isDirect();
+            return buf.isDirect();
         }
         if (buf instanceof DoubleBuffer) {
-            return ((DoubleBuffer) buf).isDirect();
+            return buf.isDirect();
         }
         if (buf instanceof LongBuffer) {
-            return ((LongBuffer) buf).isDirect();
+            return buf.isDirect();
         }

```

Comment:
And then why even have the util method at all.  (Note: it will also NPE on a null buffer.)

Reply:
```
    /*
     * FIXME when java 1.5 supprt is dropped - replace calls to this method with
     * Buffer.isDirect
     * 
     * Buffer.isDirect() is only java 6. Java 5 only have this method on Buffer
     * subclasses : FloatBuffer, IntBuffer, ShortBuffer,
     * ByteBuffer,DoubleBuffer, LongBuffer. CharBuffer has been excluded as we
     * don't use it.
     * 
     */
```
Since Java 5 is ancient history, I'll open an issue for someone to act on the above suggestion.

Reply:
opened issue #1288